lib/winapi/libloaderapi.c

```diff
@@ -30,9 +31,46 @@ BOOL FreeLibrary (HMODULE hLibModule)
     return TRUE;
 }
 
+static PIMAGE_EXPORT_DIRECTORY find_edataxb (void)
+{
+    DWORD num_sections = CURRENT_XBE_HEADER->NumberOfSections;
+    PXBE_SECTION_HEADER section_header_addr = CURRENT_XBE_HEADER->PointerToSectionTable;
+
+    for (DWORD i = 0; i < num_sections; i++) {
+        if (memcmp(section_header_addr[i].SectionName, ".edataxb", 8) == 0) {
```

Comment:
I consider this name (".edataxb") extremly ugly and hacky, and I hope it will be renamed to something better, before XBE/DXT/DLL with it start shipping.

I don't have ideas for a better name though.

Ideally it should just remain `.edata` in my opinion. Could we have a rule like:
- `-merge:.edata=.edata`
- `-merge:.edata=.edata$`
- `-merge:.edata$=.edata`.

Would those still merge to `.rdata` (or even MSVC / Clang behaviour differences maybe)?

If nothing else works, `.exports` is also a slightly better section name to me.
*(Although in a year or so, I'll probably start complaining that it is not closely related to `.edata`, which makes it harder to find the docs)*

Reply:
`-merge:.edata=.edata` works, but only with lld. It crashes the MSVC linker. The other two don't work, they cause the name to be mangled with following linker parameters.
I originally didn't want to use something that was incompatible with MSVC, because LLVM/lld might change their behavior, but I guess if this ends up being a problem, we could submit a patch upstream to introduce an xbox subsystem that will have the [default merge rule](https://github.com/llvm/llvm-project/blob/184377da5c7ce624adf09341360c6cf8f31ebe8f/lld/COFF/Driver.cpp#L1654) disabled, so I submitted #462 to change it.

Reply:
>  we could submit a patch upstream to introduce an xbox subsystem that will have the default merge rule disabled

I'd rather avoid such massive changes in a huge project for a small project like nxdk.
Disabling merge rules using a generic switch (not Xbox-specific) could be a good option though.

I wonder if there isn't a better way, like recreating this section as part of Cxbe from the EXE header (probably issues with overlapping sections or alignment when splitting into a section?).

We could also rename from `.edata` to `xbe:.edata` (= make the command part of the name) and then execute the rename in Cxbe (which runs into many section-name length conflicts anyway, so it makes sense to provide a step for it). Might trip debuggers though.

I also had the same issue with xbe-loader and to this day, I believe I hated all solutions that I've tried so far.